onnxruntime/core/providers/dml/DmlExecutionProvider/src/Operators/DmlOperatorRoiAlign.cpp

```diff
@@ -48,12 +48,15 @@ class DmlOperatorRegionOfInterestAlign : public DmlOperator, public RoiAlignHelp
         operatorDesc.MaximumSamplesPerOutput = (samplesPerOutput == 0) ? UINT32_MAX : samplesPerOutput;
         operatorDesc.ReductionFunction = *optionalReductionFunction;
         operatorDesc.InterpolationMode = DML_INTERPOLATION_MODE_LINEAR;
-        DML_OPERATOR_DESC opDesc = { DML_OPERATOR_ROI_ALIGN, &operatorDesc };
+        operatorDesc.InputPixelOffset = m_coordinateTransformMode == "half_pixel" ? 0.5f : 0.0f;
```

Comment:
We should validate the coordinate transformation mode, like what is done for `optionalReductionFunction` via the `NameAndIndex`. Otherwise if we see something we don't support, we'll just silently "support" it and return the wrong result, or we'll accept things like `"foobar"` as a legal string. Granted, opset 16 should only return the things we already know, and nobody should put bogus strings into models, and a future someone should properly update things to include any new enums added later... but it's good guard for the future.

```
constexpr NameAndIndex coordinateTransformationModes[] =
{
    {"half_pixel", 0},
    {"output_half_pixel", 1},
};

...
    std::string coordinateTransformationMode = kernelCreationContext.GetOptionalAttribute<std::string>(AttrName::CoordinateTransformationMode, "half_pixel");
    auto optionalCoordinateTransformationModeValue = TryMapStringToIndex(coordinateTransformationMode, coordinateTransformationModes);
    ML_CHECK_VALID_ARGUMENT(!!optionalCoordinateTransformationModeValue, "Unsupported RoiAlign coordinate_transformation_mode.");
...
```